utils/near-rate-limiter/README.md

```diff
@@ -0,0 +1,52 @@
+# `near-rate-limiter` Overview:
+
+Currently, in `near` we use `Actix` `Actors` multi-threaded framework for running tasks on different threads as well as 
+inter-thread communication.
+For the purpose of communicating with the outside world, we use `std::tokio::TcpStream` streams.
+Typically, we open a new connection, we define handles in actors, and `tokio` handles communication between raw Tcp sockets, and `Actix` `Actors` framework.
+Encoding/decoding messages, as well as splitting messages into frames, is done by `FramedRead`, `FramedWrite` pair from `tokio`.
+
+However, neither `tokio`, nor `actix` solve the issue of load-balancing. 
+`near-rate-limiter` provides a `ThrottledFrameRead` as a replacement for `FramedRead` with a couple of extra features.
+
+## Implemented features:
+- Stopping/resuming reading from tcp socket at will.
+- `ThrottledFramedRead` can be controlled with a helper data structure `ThrottleController`.
+- For example, can be based on total memory usage, cpu, or other metrics, `ThrottleFramedRead` can stop or resume reading from socket.
+- An `ActixMessageWrapper`, wraps around `Actix` messages, can be used to track, the number, and size of all messages
+that originated from `TcpSocket`, and are still alive, and/or being transported inside `Actix` mailboxes, etc.
+The full design, needs its own separate section.
```

Comment:
Might be helpful to create an issue for this so that we don't forget about it

Reply:
Added TODO + https://github.com/near/nearcore/issues/5672